pywbemtools/pywbemcli/_connection_repository.py

```diff
@@ -274,57 +326,49 @@ def _load_connections_file(self):
                             self._loaded = True
                     except KeyError as ke:
                         raise KeyError("Items missing from record {0} in "
-                                       "connections file. Exception {1}".format
-                                       (ke, self._connections_file))
+                                       "connections file. Exception {1}".
+                                       format(ke, self._connections_file))
                     except TypeError as te:
                         raise TypeError('Invalid object type in connections '
                                         'file: "{0}"; server name: "{1}". '
-                                        'Item: {2}'.
+                                        'Item: "{2}"'.
                                         format(self._connections_file, name,
                                                te))
-                except ValueError as ve:
-                    raise ValueError("Invalid YAML in connections file {0}. "
-                                     "Exception {1}".format
-                                     (self._connections_file, ve))
+
+                except (TypeError, yaml.YAMLError) as exc:
+                    raise TypeError('Invalid YAML in connections file: "{0}". '
+                                    "Exception: {1}".format
+                                    (self._connections_file, exc))
+
         else:
             raise ConnectionsFileNotFoundError(
                 "Connections file does not exist: {0}".
                 format(self.connections_file))
 
-    def add(self, name, svr_definition):
+    def add(self, svr_definition):
         """
         Add a new connection to the connections repository or replace an
         existing connection.  Users of this method should check before add if
         they do not want to replace an existing entry.
 
         Parameter:
 
-          svr_definition (:class:`ServerDefinition`):
-            An instance of ServerDefinition that contains the data that will be
-            added to the connections file.
+          svr_definition (:class:`PywbemServer`):
+            An instance of PywbemServer that contains the data that will be
+            added to the connections file. The server definition must contain
+            a name attribute.
 
         Raises:
-            click.Abort if the load of the connections file returns an
-            exception
+            ValueError
+            TypeError
```

Comment:
It can also raise IOError, KeyError and ConnectionsFileNotFoundError.

Reply:
**Update:** I'll improve the exception handling of `ConnectionRepository` after this PR is merged, so no need to invest in that as part of this PR. You can ignore my comment above.

Reply:
**Update:** I'll improve the exception handling of `ConnectionRepository` after this PR is merged, so no need to invest in that as part of this PR. You can ignore my comment above.

Reply:
DONE